maraboupy/MarabouNetworkNNet.py

```diff
@@ -163,66 +375,305 @@ def variableRanges(self):
+
+        Returns:
+            (list of float)
+        """
+        bound_list = [self.getLowerBound(layer, node, b) for node in range(self.layerSizes[layer])]
+        return bound_list
+
+    def getBoundsForLayer(self, layer, b=True):
+        """ Returns a tuple of two lists, the lower and upper bounds for the variables corresponding to the given layer
+
+            If b=True, b variable
+            Otherwise, f variable
+
+        Args:
+            layer (int): Layer number
+            b (bool): If True, the b variable is considered, otherwise the f variable
+
+        Returns:
+            (tuple of two lists of float): list of lower bounds, list of upper bounds
+        """
+        return self.getLowerBoundsForLayer(layer, b), self.getUpperBoundsForLayer(layer, b)
+
+    def numberOfVariables(self):
+        """Get total number of variables in network
+
+        Returns:
+            (int)
+        """
+        return self.layerSizes[0] + 2 * sum(self.layerSizes[1:-1]) + 1 * self.layerSizes[-1]
+
+    def getInputMinimum(self, input_var):
+        """Get minimum value for given input variable
+
+        Args:
+            input_var (int): Index of input variable
+
+        Returns:
+            (float)
+        """
+        return self.inputMinimums[input_var]
+
+    def getInputMaximum(self, input_var):
+        """Get maximal value for given input variable
+
+        Args:
+            input_var (int): index of input variable
+
+        Returns:
+            (float)
+        """
+        return self.inputMaximums[input_var]
+
+    def evaluateWithoutMarabou(self, inputValues):
+        """ Evaluate network directly (without Marabou) at a given point
+
+        Args:
+            inputValues (list of np array): Input to network
+
+        Returns:
+            (np array): Output of the network
+        """
+        return self.evaluateNNet(inputValues.flatten().tolist(), normalize_inputs=self.normalize,
+                                 normalize_outputs=self.normalize)
+
+    def evaluateNNet(self, inputs, first_layer = 0, last_layer=-1, normalize_inputs=False,
+                     normalize_outputs=False, activate_output_layer=False):
+        """ Evaluate nnet directly (without Marabou) at a given point
+
+        Evaluates the network on input inputs between layer first_layer (input layer by default)
+        and layer last_layer (output layer by default)
+
+        Args:
+            inputs: (list of float):         Network inputs to be evaluated
+            first_layer:  (int):              The initial layer of the evaluation
+            last_layer: (int):                The last layer of the evaluation
+            normalize_inputs: (bool):         If True and first_layer==0, normalization of inputs is performed
+            normalize_outputs: (bool):        If True, normalization of output is undone
+            activate_output_layer: (bool):    If True, the last layer is activated, otherwise it is not.
```

Comment:
Please remove the colon after parameter names. There should only be a colon after the (parameter type).

Docstring syntax error like this can be found by building the documentation locally, following the instructions in the maraboupy/docs folder. I did this just now and saw that the parameter formatting for this function was a little off. Once this is fixed and we merge into master, [this website](https://neuralnetworkverification.github.io/Marabou/API/2_MarabouNetworkNNet.html) will automatically be updated with the new functions, class attributes, and with your name in the contributor section, so it's best practice to build the documentation locally and make sure that the documentation looks consistent. :) 

Everything else looks good, so I'm happy to merge once this is fixed. Thanks for your work!

Reply:
Thank you. Sorry this review must have involved quite a bit of work..